packages/rsocket-core/src/RequestResponseRequesterStream.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  CancelFrame,
+  Cancellable,
+  ErrorFrame,
+  ExtFrame,
+  Flags,
+  FrameTypes,
+  Outbound,
+  Payload,
+  PayloadFrame,
+  RequestNFrame,
+  RequestResponseFrame,
+  RSocketError,
+  StreamConfig,
+  StreamFrameHandler,
+  StreamLifecycleHandler,
+  StreamsRegistry,
+  UnidirectionalStream,
+} from "@rsocket/rsocket-types";
+import { fragment, isFragmentable } from "./Fragmenter";
+import * as Reassembler from "./Reassembler";
+
+export class RequestResponseRequesterStream
+  implements
+    Cancellable,
+    StreamFrameHandler,
+    StreamLifecycleHandler,
+    Reassembler.FragmentsHolder {
+  private done: boolean;
+  private outbound: Outbound;
+  private fragmentSize: number;
+
+  hasFragments: boolean;
+  data: Buffer;
+  metadata: Buffer;
+
+  streamId: number;
+
+  constructor(
+    private payload: Payload,
+    private receiver: UnidirectionalStream,
+    private streamsRegistry: StreamsRegistry
+  ) {
+    streamsRegistry.add(this);
+  }
+
+  handleReady(
+    streamId: number,
+    { outbound, fragmentSize }: StreamConfig
+  ): boolean {
+    if (this.done) {
+      return false;
+    }
+
+    this.streamId = streamId;
+    this.outbound = outbound;
+    this.fragmentSize = fragmentSize;
+
+    if (
+      !isFragmentable(this.payload, fragmentSize, FrameTypes.REQUEST_RESPONSE)
+    ) {
+      this.outbound.send({
+        type: FrameTypes.REQUEST_FNF,
+        data: this.payload.data,
+        metadata: this.payload.metadata,
+        flags: this.payload.metadata ? Flags.METADATA : 0,
+        streamId,
+      });
+    } else {
+      for (const frame of fragment(
+        streamId,
+        this.payload,
+        fragmentSize,
+        FrameTypes.REQUEST_RESPONSE
+      )) {
+        this.outbound.send(frame);
+      }
+    }
+
+    return true;
+  }
+
+  handleReject(error: Error): void {
+    if (this.done) {
+      return;
+    }
+
+    this.done = true;
+
+    if (this.hasFragments) {
+      Reassembler.cancel(this);
+    }
+
+    this.receiver.onError(error);
+  }
+
+  handle(
+    frame: PayloadFrame | ErrorFrame | CancelFrame | RequestNFrame | ExtFrame
+  ): void {
+    if (this.done) {
+      return;
+    }
+
+    switch (frame.type) {
+      case FrameTypes.PAYLOAD:
```

Comment:
nit: Could extract the individual case bodies to their own methods to reduce the length of the switch statement and nesting in this method.